assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -62,14 +66,23 @@ const RightMenu: React.FC<any> = () => {
   const screenPropertyChangeHandler = useCallback(
     (properties: any) => {
       if (currentActivity) {
+        const currentSequence = findInSequence(sequence, currentSequenceId);
         const modelChanges = transformScreenSchematoModel(properties);
+        //currentSequence?.custom.sequenceName = modelChanges.title;
         console.log('Screen Property Change...', { modelChanges });
+        const title = modelChanges.title;
+        delete modelChanges.title;
         const screenChanges = {
           ...currentActivity?.content?.custom,
           ...modelChanges,
         };
         const cloneActivity = clone(currentActivity);
         cloneActivity.content.custom = screenChanges;
+        if(title){
+          cloneActivity.title = title;
+          cloneActivity.activitySlug = title.toLowerCase().split(' ').join('_');
```

Comment:
the slug should be generated on the server, need to check if the new slug comes back from the save (edit) call or not. if it does, then we need to update the redux item in `saveActivity` if it doesn't then will need to look into adding that server side I think